shell/packages/sandstorm-db/db.js

```diff
@@ -1810,23 +1810,81 @@ if (Meteor.isServer) {
   // Cleanup tokens every hour.
   SandstormDb.periodicCleanup(3600000, cleanupExpiredAssetUploads);
 
-  SandstormDb.prototype.getIconUrls = function (icon) {
-    check(icon, Match.OneOf(undefined, null,
-                            { format: String, assetId: String,
-                              assetId2xDpi: Match.Optional(String), }));
-    const root =
-        Url.parse(process.env.ROOT_URL).protocol + "//" + this.makeWildcardHost("static") + "/";
+  SandstormDb.prototype.updateCachedViewInfo = function (grainId, viewInfo) {
```

Comment:
Probably worth adding some doc comment explaining what this method is doing.

I guess all the refcounting here only happens if the app defines grain-specific overrides for the icons, which it usually doesn't, right?